woocommerce/woocommerce-ios
# Fix in-stock product variations with "Manage stock" disabled showing wrong stock status in variations list

Fixes #2934 

## Changes

- In `ProductFormDataModel`'s extension helper `createStockText` that generates the text for stock details, it only shows the non-nil stock quantity if manage stock is enabled (when it is disabled, the stock quantity is invalid even though it could be non-nil).
- Added/updated unit tests

## Testing

- Go to the Products tab
- Tap on a variable product
- Tap on the variations row
- Tap on a variation
- Tap on the inventory row
- Turn off "Manage stock", and set "Stock status" to "In stock"
- Tap "Done" to save the change to the Inventory settings
- Tap "Update" on the variation to save it remotely
- Go back to the variations list --> the variation row should read "In stock" without a quantity

## Example screenshots

before | after
-- | --
![Simulator Screen Shot - iPhone 11 - 2020-10-08 at 15 21 24](https://user-images.githubusercontent.com/1945542/95437629-cc6f1900-0987-11eb-9a00-08b4a31098f2.png) | ![Simulator Screen Shot - iPhone 11 - 2020-10-08 at 15 25 02](https://user-images.githubusercontent.com/1945542/95437647-d09b3680-0987-11eb-9eaf-639a809d1d7d.png)


Update release notes:
